### ISSUE TITLE
Custom migration name

### ISSUE THREAD

**Issue by joshkel** [knex/knex#6012]
The section on ["Custom migration name"](https://knexjs.org/guide/migrations.html#custom-migration-name) describes the `getNewMigrationName` option, but that feature has not been merged: https://github.com/knex/knex/pull/5422

**Comment by DarkRoku12** [comment]
I also have this issue, I think that as a workaround, you may use a script that auto-generates it.

**Comment by xpuu** [comment]
Please merge this <3

**Comment by mercmobily** [comment]
#5422 feels like feature creep. The current timestamp-based naming works well and follows industry standards. Custom naming adds complexity for minimal benefit. Most teams needing special naming can simply rename files after generation or wrap the Knex CLI.

To close this issue we should fix the documentation so that it doesn't describe a feature that does not exist.

**Comment by joshkel** [comment]
Thanks for following up, @mercmobily. For background information, we were interested in this feature because we wanted to enforce some standardization on our migration names (timestamp plus snake_case, to prevent someone from accidentally using, e.g., timestamp plus kebab-case). We ended up using [eslint-plugin-filenames](https://www.npmjs.com/package/eslint-plugin-filenames) to enforce this standardization outside of Knex.

**Comment by mkicks** [comment]
From my point of view, adding an opinionated default and an option to use a callback for customization always makes sense. Besides, for smaller teams (the majority of users), a long name for a migration just doesn’t give any benefit. The date is only for ordering, while the meaningful information is in the migration name, which is hardly visible in the IDE explorer or tabs.

I do, however, understand the need to clear the clutter of unresolved issues as quickly as possible. But this has already been fixed - it just needs to be merged.

**Comment by xpuu** [comment]
From my point of view, adding an opinionated default and an option to use a callback for customization always makes sense. Besides, for smaller teams (the majority of users), a long name for a migration just doesn’t give any benefit. The date is only for ordering, while the meaningful information is in the migration name, which is hardly visible in the IDE explorer or tabs.

I do, however, understand the need to clear the clutter of unresolved issues as quickly as possible. But this has already been fixed - it just needs to be merged.

**Comment by mercmobily** [comment]
You are right, and I agree. I changed this into "bug", and the merging will fix it. This should have been the response in the first place. Thank you for straightening the course.

**Comment by xpuu** [comment]
@mercmobily Many thanks for keeping the project alive 🙏❤️

**Comment by mercmobily** [comment]
Oh more than alive. Let me finish this huge triaging exercise (1100 issues to be checked, ONE BY ONE) and then you will see... I am sorry I dropped the ball on this one actually.
